packages/compass-collection/src/components/collection-header/collection-header.tsx

```diff
@@ -1,33 +1,77 @@
+/* eslint-disable jsx-a11y/anchor-is-valid */
+import type AppRegistry from 'hadron-app-registry';
+import { css } from '@mongodb-js/compass-components';
+import type { Document } from 'mongodb';
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 import toNS from 'mongodb-ns';
 
 import CollectionHeaderActions from '../collection-header-actions';
 import ReadOnlyBadge from './read-only-badge';
 import TimeSeriesBadge from './time-series-badge';
 import ViewBadge from './view-badge';
 
-import styles from './collection-header.module.less';
+const collectionHeaderStyles = css({
+  paddingTop: '15px',
+  paddingBottom: '5px',
+});
 
-class CollectionHeader extends Component {
-  static displayName = 'CollectionHeaderComponent';
+const collectionHeaderNamespaceStyles = css({
+  display: 'contents',
+});
 
-  static propTypes = {
-    globalAppRegistry: PropTypes.object.isRequired,
-    namespace: PropTypes.string.isRequired,
-    isReadonly: PropTypes.bool.isRequired,
-    isTimeSeries: PropTypes.bool.isRequired,
-    statsPlugin: PropTypes.func.isRequired,
-    selectOrCreateTab: PropTypes.func.isRequired,
-    statsStore: PropTypes.object.isRequired,
-    sourceName: PropTypes.string,
-    sourceReadonly: PropTypes.bool.isRequired,
-    sourceViewOn: PropTypes.string,
-    editViewName: PropTypes.string,
-    pipeline: PropTypes.array
-  };
+const collectionHeaderTitleStyles = css({
+  fontSize: '24px',
+  fontWeight: 'normal',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  display: 'flex',
+  paddingLeft: '15px',
+  paddingRight: '15px',
+  margin: 0,
+  lineHeight: '32px',
+  alignItems: 'center',
+});
+
+const collectionHeaderTitleDBStyles = css({
+  color: ' #337ab7',
+  flexShrink: 2,
+  flexBasis: 'auto',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+  cursor: 'pointer',
+  textDecoration: 'none',
+  '&:hover,&:focus': {
+    textDecoration: 'underline',
+  },
+});
+
+const collectionHeaderTitleCollectionStyles = css({
+  flexBasis: 'auto',
+  overflow: 'hidden',
+  textOverflow: 'ellipsis',
+  whiteSpace: 'nowrap',
+});
+
+type CollectionHeaderProps = {
+  globalAppRegistry: AppRegistry;
+  namespace: string;
+  isReadonly: boolean;
+  isTimeSeries: boolean;
+  statsPlugin: React.FunctionComponent<any>;
```

Comment:
I think it’s fine to be a tiny bit more specific here, feel free to ignore:

```suggestion
  statsPlugin: React.FunctionComponent<{ store: any }>;
```